### ISSUE TITLE
About the deeplabv2 performance under full supervision

### ISSUE THREAD

**Issue by liberty523422842** [jbeomlee93/BBAM#1]
I've tried to reproduce the semantic segmentation results on PASCAL VOC12 dataset under full supervision, using code in https:// github.com/kazuto1011/deeplab-pytorch and training settings in your paper, initialized from the ImageNet pretrained model, but got 74.2 (mIoU), not 76.8 in your paper. I wonder how your model initializes. 

Wish you release your code soon. Thank you.

**Comment by TyroneLi** [comment]
> I've tried to reproduce the semantic segmentation results on PASCAL VOC12 dataset under full supervision, using code in https:// github.com/kazuto1011/deeplab-pytorch and training settings in your paper, initialized from the ImageNet pretrained model, but got 74.2 (mIoU), not 76.8 in your paper. I wonder how your model initializes.
> 
> Wish you release your code soon. Thank you.

You may try to initialize with COCO pretrained. They may do like this but comment different.

**Comment by jbeomlee93** [comment]
Hi @liberty523422842, I have conducted all the experiments using an ImageNet pre-trained model (https://drive.google.com/file/d/17-B3wPBYRYN2_hdca5B3zGfKQiCg32W0/view?usp=sharing).

Using a COCO pre-trained model provided by the original repository, I obtained 76.73 mIoU before CRF, and 78.16 mIoU after CRF.

Using an ImageNet pre-trained model I referred to above, I obtained 73.37 mIoU before CRF, and 76.77 mIoU after CRF. 


Thanks.

**Comment by liberty523422842** [comment]
> Hi @liberty523422842, I have conducted all the experiments using an ImageNet pre-trained model (https://drive.google.com/file/d/17-B3wPBYRYN2_hdca5B3zGfKQiCg32W0/view?usp=sharing).
> 
> Using a COCO pre-trained model provided by the original repository, I obtained 76.73 mIoU before CRF, and 78.16 mIoU after CRF.
> 
> Using an ImageNet pre-trained model I referred to above, I obtained 73.37 mIoU before CRF, and 76.77 mIoU after CRF.
> 
> Thanks.

Thank you for your pretrain model !!!
I'll try later.

**Comment by liberty523422842** [comment]
while using the pretrained model you shared with the default training setting in kazuto1011'repo. I got a mIoU of 76.63 before CRF, and 78.07 after CRF. Is it too high for a ImageNet pretrained model ?
<img width="483" alt="截屏2021-04-14 上午1 08 21" src="https://user-images.githubusercontent.com/22132742/114592796-e9c9ca80-9cbd-11eb-8832-14ebfa648596.png">
<img width="257" alt="截屏2021-04-14 上午1 08 47" src="https://user-images.githubusercontent.com/22132742/114592856-f9491380-9cbd-11eb-93b7-0557e0f75d85.png">
<img width="323" alt="截屏2021-04-14 上午1 07 13" src="https://user-images.githubusercontent.com/22132742/114592665-c0a93a00-9cbd-11eb-9dd4-4cc042f37b6e.png">
<img width="335" alt="截屏2021-04-14 上午1 07 25" src="https://user-images.githubusercontent.com/22132742/114592695-c99a0b80-9cbd-11eb-987a-454b1c2b4e45.png">

**Comment by TyroneLi** [comment]
> Hi @liberty523422842, I have conducted all the experiments using an ImageNet pre-trained model (https://drive.google.com/file/d/17-B3wPBYRYN2_hdca5B3zGfKQiCg32W0/view?usp=sharing).
> 
> Using a COCO pre-trained model provided by the original repository, I obtained 76.73 mIoU before CRF, and 78.16 mIoU after CRF.
> 
> Using an ImageNet pre-trained model I referred to above, I obtained 73.37 mIoU before CRF, and 76.77 mIoU after CRF.
> 
> Thanks.

When do u plan to release code?

**Comment by liberty523422842** [comment]
> > Hi @liberty523422842, I have conducted all the experiments using an ImageNet pre-trained model (https://drive.google.com/file/d/17-B3wPBYRYN2_hdca5B3zGfKQiCg32W0/view?usp=sharing).
> > Using a COCO pre-trained model provided by the original repository, I obtained 76.73 mIoU before CRF, and 78.16 mIoU after CRF.
> > Using an ImageNet pre-trained model I referred to above, I obtained 73.37 mIoU before CRF, and 76.77 mIoU after CRF.
> > Thanks.
> 
> When do u plan to release code?

my code ？For mIoU of 74.2. You can use kazuto1011'repo and ImageNet pretrained ResNet-101 produced by pytorch. But you  need to pay attention to the image processing they used (https://pytorch.org/hub/pytorch_vision_resnet).

**Comment by liberty523422842** [comment]
@jbeomlee93 I've tried to reproduce your result with your pretrained model. I got mIoU of 73.95 before CRF which is higher than your result, but 75.72 after CRF. There is still a gap with your result. Have you changed the CRF parameters ?

**Comment by jbeomlee93** [comment]
@liberty523422842 

I am trying to reproduce the results with ImageNet pre-trained model.

It not quite accurate, but I think...

1) I have changed the learning rate because a value of 2.5e-4 makes "nan" loss. I think it was 2.0e-4 or lower.

2)  I have changed augmentation scales for training, maybe from [0.5, 0.75, 1.0, 1.25, 1.5] to [0.5, 0.75, 1.0, 1.25, 1.5, 1.75, 2.0]

3) I have changed augmentation scales in inference time, from [0.5, 0.75, 1.0] to [0.5, 0.75, 1.0, 1.25]

4) I have not changed any CRF parameters.

Please note that, I have obtained the pre-trained model by converting ImageNet pre-trained caffe model with convert.py in this repo [https://github.com/kazuto1011/deeplab-pytorch].

Thanks.
